thinc/loss.py

```diff
@@ -126,12 +142,21 @@ def convert_truths(self, truths, guesses: Floats2d) -> Tuple[Floats2d, Floats2d]
     def __call__(
         self, guesses: Floats2d, truths: IntsOrFloatsOrStrs
     ) -> Tuple[Floats2d, float]:
+        # XXX not ideal to run convert_input and _validate_input twice.
```

Comment:
Maybe it's useful to check the cost. If it's too much, I guess an alternative could be to have `_get_grad` and `_get_loss` private methods that don't do the validation and conversion, with would then be called by the entry points `__call__`, `get_grad` and `get_loss`, so we'd only have to do validation once for `__call__`.